src/lib/evil/evil_libgen.h

```diff
@@ -0,0 +1,13 @@
+#ifndef __EVIL_LIBGEN_H__
+#define __EVIL_LIBGEN_H__
+
+#include <evil_private.h>
+
+#ifdef _MSC_VER
+# define dirname evil_dirname
+#endif
```

Comment:
I bring the same question that I brought at `evil_basename`, should we use [_MSC_VER or HAVE_FOO](https://github.com/expertisesolutions/efl/pull/323#discussion_r458078126) and why?